examples/35_gemm_softmax/gemm_online_softmax.cpp

```diff
@@ -0,0 +1,604 @@
+      cutlass::TensorView<ElementC, LayoutC> view_Ref_device(block_ref_D.get(), layout_C, extent_C);
+
+      cutlass::reference::device::GemmComplex<
+          ElementA, LayoutA,
+          ElementB, LayoutB,
+          ElementC, LayoutC, 
+          ElementCompute, ElementCompute, ElementD
+      >(
+        problem_size,
+        options.alpha, 
+        view_A,
+        cutlass::ComplexTransform::kNone,
+        view_B,
+        cutlass::ComplexTransform::kNone,
+        options.beta, 
+        view_C, 
+        view_Ref_device, 
+        ElementCompute(0)
+      );
+
+      // Copy reference results to host memory for verification
+      std::vector<ElementD> matrix_D_Ref(layout_C.capacity(extent_C));
+      cutlass::device_memory::copy_to_host(matrix_D_Ref.data(), block_ref_D.get(), matrix_D_Ref.size());
+      cutlass::TensorView<ElementD, LayoutC> view_D_Ref(matrix_D_Ref.data(), layout_C, extent_C);
+
+      std::vector<ElementSoftmax> matrix_Softmax_Ref(layout_C.capacity(extent_C));
+      cutlass::TensorView<ElementSoftmax, LayoutC> view_Softmax_Ref(matrix_Softmax_Ref.data(), layout_C, extent_C);
+
+      // Copy computed results to host memory
+      std::vector<ElementD> matrix_D(layout_C.capacity(extent_C));
+      cutlass::device_memory::copy_to_host(matrix_D.data(), block_D.get() + total_elements_D_per_batch * batch_idx, matrix_D.size());
+
+      // Compute the norm
+      for (int m = 0; m < options.m; ++m) {
+        reference_N.at({m, 0}) = view_D_Ref.ref().at({m, 0});
+        for (int n = 1; n < options.n; ++n) {
+          reference_N.at({m, 0}) = std::max(reference_N.at({m, 0}), ElementSoftmax(view_D_Ref.ref().at({m, n})));
+        }
+      }
+
+      // Compute softmax
+      for (int m = 0; m < options.m; ++m) {
+        float sum = 0;
+        for (int n = 0; n < options.n; ++n) {
+          sum += std::exp( float(view_D_Ref.ref().at({m, n})) - float(reference_N.at({m, 0})) );
+        }
+        float inv_sum = float(1.0f / sum);
+
+        for (int n = 0; n < options.n; ++n) {
+          view_Softmax_Ref.ref().at({m, n}) = ElementSoftmax(
+            std::exp( float(view_D_Ref.ref().at({m, n})) - float(reference_N.at({m, 0})) ) * inv_sum
+          );
+        }
+      }
+
+      bool verified_Softmax = verify_tensor<ElementSoftmax>(matrix_D, matrix_Softmax_Ref, options);
+      if (!verified_Softmax) {
+        std::cerr << "Verification of Softmax tensor failed\n";
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /// Initialize operands to be used in the GEMM and reference GEMM
+  void initialize(const ProblemShapeType& problem_size) {
+    auto problem_shape_MNKL = cute::append<4>(problem_size, 1);
+    auto [M, N, K, L] = problem_shape_MNKL;
+
+    auto partials_N = cute::ceil_div(N, cute::shape<1>(typename Gemm::TileShape{}));
+    auto tmp_size = M * partials_N * L;
```

Comment:
What does tmp mean? Is this for the partial reduction? If so, should we use part_reduction (or something similar) as a name?

Same for stride_tmp and layout_tmp.

Reply:
fixed